crates/telio-dns/tests/nameserver.rs

```diff
@@ -433,7 +434,7 @@ async fn dns_request_forward_to_slow_server() {
     // Start a query that will run for several seconds
     tokio::spawn(dns_test_with_server(
         "google.com",
-        DnsTestType::CorrectIpv4,
+        DnsTestType::NonRespondingForwardServer,
```

Comment:
Is this test still testing something? It should verify that we can modify the nameserver while the query **is still running**

Reply:
As far as I understand, it validates that the nameserver is still available while having still running queries. Please, correct me if I'm wrong.
And yes, it makes sense to me to also validate if we can modify the nameserver while the query is still running, but I would prefer to do it in a different PR (for extending tests on telio-dns).

Reply:
Sorry if I wasn't clear. I'm slightly worried that after the change to the underlying trust-dns library we no longer have a long running query here since it is 'immediately' fails with that new error. If you are confident that this is not the case feel free to resolve this conversation :)